Services/UI/OJS.Services.Ui.Business/Validations/Implementations/Contests/ContestParticipationValidationService.cs

```diff
@@ -1,53 +1,60 @@
 ﻿namespace OJS.Services.Ui.Business.Validations.Implementations.Contests;
 
-using SoftUni.AutoMapper.Infrastructure.Extensions;
 using OJS.Data.Models.Contests;
-using OJS.Services.Common.Models.Contests;
 using OJS.Services.Common;
 using OJS.Services.Common.Models;
+using OJS.Services.Common.Models.Contests;
 using OJS.Services.Common.Models.Users;
 using OJS.Services.Infrastructure;
+using OJS.Services.Ui.Data;
+using SoftUni.AutoMapper.Infrastructure.Extensions;
 
-public class ContestValidationService : IContestValidationService
+public class ContestParticipationValidationService : IContestParticipationValidationService
 {
     private readonly IDatesService datesService;
     private readonly IContestsActivityService activityService;
     private readonly IContestCategoriesBusinessService categoriesService;
+    private readonly ILecturersInContestsBusinessService lecturersInContestsBusiness;
+    private readonly IContestsDataService contestsData;
 
-    public ContestValidationService(
+    public ContestParticipationValidationService(
         IDatesService datesService,
         IContestsActivityService activityService,
-        IContestCategoriesBusinessService categoriesService)
+        IContestCategoriesBusinessService categoriesService,
+        ILecturersInContestsBusinessService lecturersInContestsBusiness,
+        IContestsDataService contestsData)
     {
         this.datesService = datesService;
         this.activityService = activityService;
         this.categoriesService = categoriesService;
+        this.lecturersInContestsBusiness = lecturersInContestsBusiness;
+        this.contestsData = contestsData;
     }
 
     public ValidationResult GetValidationResult((Contest?, int?, UserInfoModel?, bool) item)
     {
         var (contest, contestId, user, official) = item;
 
         // TODO: Fix so it uses lecturers in contests business service
-        var isUserLecturerInContest = contest != null && user != null && user.IsLecturer;
+        var userIsAdminOrLecturerInContest = user != null && (this.lecturersInContestsBusiness
+            .IsCurrentUserAdminOrLecturerInContest(contest?.Id)
+            .GetAwaiter()
+            .GetResult() || user.IsAdmin);
 
         if (contest == null ||
-            user == null ||
```

Comment:
Better leave this check here, so we don't make extra db requests in the following checks.